### PR TITLE
#Issue 27715: Implement RegPrints Class for ARM64 SIMD Platforms

### PR DESCRIPTION
**Pull request by amanmogal**
Details:

 New File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/debug_capabilities.hpp
Class: RegPrints

Methods:

print_gp_registers: Prints the contents of general-purpose registers.

print_vector_registers: Prints the contents of vector registers.

usage_description: Provides a detailed explanation of how to use the RegPrints class for debugging.

Helper Methods:

print_gp_register: Helper function to print a single general-purpose register.

print_vector_register: Helper function to print a single vector register.

Functionality:

The RegPrints class includes methods to print the contents of both general-purpose and vector registers.

This enables developers to debug the state of registers during JIT kernel execution on ARM64 SIMD platforms.

The class includes a description of its usage for better understanding and implementation.

### Tickets:
Issue https://github.com/openvinotoolkit/openvino/issues/27715

### CODE REVIEW

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/debug_capabilities.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    static void print_gpr(const uint64_t& reg_value, const char* reg_name) {
+        std::cout << "Register " << reg_name << ": " << std::hex << reg_value << std::endl;
+    }
+
+    // Print vector registers (SIMD)
+    static void print_simd(const float32x4_t& reg_value, const char* reg_name) {
+        float values[4];
+        vst1q_f32(values, reg_value); // Store SIMD register into an array
+        std::cout << "SIMD Register " << reg_name << ": ["
+                  << values[0] << ", " << values[1] << ", "
```
Comment (praasz) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/debug_capabilities.hpp:20]: Detail:
use `ov::util::join(values)`

File: .vscode/settings.json
Comment (praasz) [inline .vscode/settings.json:1]: Remove this file from PR?

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/debug_capabilities.hpp
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/debug_capabilities.hpp:1]: Hi @amanmogal , thanks for the contribution.
Your impl seems to be different with what is expected (see x64 impl https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/emitters/plugin/x64/debug_capabilities.hpp). 
You need to understand that there are 2 phases during interaction with JIT code generation:
1. Code generation phase (then print_gpr and print_simd functions are called)
2. Code execution stage (then generated code is called).
In other words print_gpr and print_simd functions should generate code, that will print registers during code execution stage. In your case std::cout tries to print smt on code generation stage, bot execution.

Comment (amanmogal) [reply src/plugins/intel_cpu/src/emitters/plugin/aarch64/debug_capabilities.hpp:1]: ok ok got it..

File: src/plugins/intel_cpu/src/emitters/plugin/aarch64/debug_capabilities.hpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#include <arm_neon.h> // For SIMD support
+#include "openvino/util/ov_string_utils.hpp" // For ov::util::join
+
+class RegPrints {
+public:
+    static void print_gpr(jit_generator &gen, const uint64_t &reg_value, const char *reg_name) {
+        // Emit JIT code to print general-purpose register during runtime
+        gen.mov(gen.rdi, reg_value); // Move register value into rdi
+        gen.mov(gen.rsi, reinterpret_cast<uint64_t>(reg_name)); // Pass register name as argument
+        gen.call(reinterpret_cast<void(*)(const char*, uint64_t)>(print_runtime_gpr)); // Call runtime function
```
Comment (a-sidorova) [inline src/plugins/intel_cpu/src/emitters/plugin/aarch64/debug_capabilities.hpp:15]: The instruction `call` is missed on aarch64. Please take a look at the instruction [blr](https://developer.arm.com/documentation/ddi0602/2024-03/Base-Instructions/BLR--Branch-with-Link-to-Register-?lang=en). It calls a subroutine at an address in a register.

The registers `rdi` and `rsi` are also missed on aarch64. This naming is used on x64 platforms.

Also, as mentioned in the issue, the perfect example for this task is the implementation of `jit_power_static_emitter` which generates code to call `std::powf` in runtime. Please take a look at this [impl](https://github.com/openvinotoolkit/openvino/blob/91a5518347a5e4e55f940b6e3c9b8f00626f09ff/src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_eltwise_emitters.cpp#L1769-L1793).
